File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/fragments/investigation/mapsmenu/mapdisplay/io/MapFileIO.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.mapsmenu.mapdisplay.io;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.mapsmenu.mapdisplay.io.models.WorldMapWrapper;

import java.io.InputStream;

public class MapFileIO {

    public WorldMapWrapper mapsWrapper = new WorldMapWrapper();

    public MapFileReader reader = new MapFileReader(mapsWrapper);

    public boolean readFile(InputStream inputStream, MapFileReader reader) {
        return reader.loadFile(inputStream);
    }

}

File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/models/maps/io/MapFileIO.kt
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.models.maps.io

import com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.models.maps.io.factory.MapDesBlueprint
import java.io.InputStream

class MapFileIO {

    private val deserializedData: MapDesBlueprint =
        MapDesBlueprint()

    val reader: MapFileReader =
        MapFileReader(
            deserializedData
        )

    fun readFile(inputStream: InputStream?, reader: MapFileReader): Boolean {
        return reader.loadFile(inputStream)
    }
}
